web_tree_many2one_clickable/static/src/js/web_tree_many2one_clickable.js

```diff
@@ -1,104 +1,112 @@
-            }
-        },
+odoo.define('web_tree_many2one_clickable.many2one_clickable', function(require) {
+"use strict";
 
-        get_options: function() {
-            if (_.isUndefined(this.ir_option_clickable_loaded)) {
-                var self = this; // Needed for binding the instance
-                this.ir_option_clickable_loaded = $.Deferred();
-                this.use_many2one_clickable = false;
-                (new instance.web.Model("ir.config_parameter"))
-                    .query(["value"])
-                    .filter([['key', '=', 'web_tree_many2one_clickable.default']])
-                    .first()
-                    .then(function(param) {
-                        if (param) {
-                            self.use_many2one_clickable = (param.value.toLowerCase() == 'true');
-                        }
-                        self.ir_option_clickable_loaded.resolve();
-                    });
-                return this.ir_option_clickable_loaded;
-            }
-            return $.when();
-        },
+var _ = require('_');
+var $ = require('$');
+
+var core = require('web.core');
+var ListView = require('web.ListView');
+var Model = require('web.DataModel');
 
-        _format: function (row_data, options)
-        {
-            if (this.use_many2one_clickable) {
-                return _.str.sprintf('<a class="oe_form_uri" data-many2one-clickable-model="%s" data-many2one-clickable-id="%s">%s</a>',
-                    this.relation,
-                    row_data[this.id].value[0],
-                    _.escape(row_data[this.id].value[1] || options.value_if_empty));
+var _t = core._t;
+var QWeb = core.qweb;
+var list_widget_registry = core.list_widget_registry;
+
+var use_many2one_clickable_default = false;
+var ir_option_clickable_loaded;
+
+function get_options() {
+    if (_.isUndefined(ir_option_clickable_loaded)) {
+        ir_option_clickable_loaded = $.Deferred();
+        (new Model("ir.config_parameter"))
+            .query(["value"])
+            .filter([['key', '=', 'web_tree_many2one_clickable.default']])
+            .first()
+            .then(function(param) {
+                if (param) {
+                    use_many2one_clickable_default = (
+                        param.value.toLowerCase() == 'true');
+                }
+                ir_option_clickable_loaded.resolve();
+            });
+        return ir_option_clickable_loaded;
+    }
+    return $.when();
+}
+
+ListView.Column.include({
+    init: function(id, tag, attrs) {
+        this._super(id, tag, attrs);
+        if (this.widget == 'many2one_clickable') {
+            this.use_many2one_clickable = true;
+        // } else if (this.widget == 'many2one_unclickable') {
+        //     this.use_many2one_clickable = false;
+        } else if (this.type == 'many2one') {
+            if (use_many2one_clickable_default) {
+                this.use_many2one_clickable = true;
+            } else {
+                get_options();
```

Comment:
it takes care that `this` in the function points to the object from the outer `this`